src/item/HorseArmor.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+use pocketmine\color\Color;
+use pocketmine\nbt\tag\CompoundTag;
+use pocketmine\nbt\tag\IntTag;
+use pocketmine\utils\Binary;
+
+class HorseArmor extends Item{
+
+	public const TAG_CUSTOM_COLOR = "customColor"; //TAG_Int
+
+	protected ?Color $customColor = null;
+
+	public function getMaxStackSize() : int{
+		return 1;
+	}
+
+	/**
+	 * Returns the dyed colour of this armour piece. This generally only applies to leather armour.
+	 */
+	public function getCustomColor() : ?Color{
+		return $this->customColor;
+	}
+
+	/**
+	 * Sets the dyed colour of this armour piece. This generally only applies to leather armour.
+	 *
+	 * @return $this
+	 */
+	public function setCustomColor(?Color $color) : self{
+		if($this->getTypeId() === ItemTypeIds::LEATHER_HORSE_ARMOR){
+			$this->customColor = $color;
+		}
+		return $this;
```

Comment:
`throw new Expection("Custom color can only be applied on Leather armor")`

Smth like this?

Reply:
maybe?

Reply:
Maybe create a separate class for leather horse armour?

Reply:
I think it's a bad idea to have a separate class just because of the custom color, there are enough anyway.

but that's not for me to decide, I'd like to have a second opinion on it

Reply:
I could rename this file to LeatherHorseArmor, the other armors doesn't need the class anyway.

Is there a way to add a item in vanillablocks with maxstacksize 1 without to create a new class?

Reply:
After checked the actual Armor class, it seems an unnecessary headache to ban coloring for non leather

Reply:
yup, i think it's better, throw exception if non leather